src/sdam/state/mod.rs

```diff
@@ -557,23 +594,27 @@ impl TopologyState {
 /// handshake for the conection being used in that operation.
 ///
 /// This is used to determine the error handling semantics for certain error types.
+#[derive(Debug, Clone)]
 pub(crate) enum HandshakePhase {
-    /// Describes an point that occurred before the handshake completed (e.g. when opening the
-    /// socket or while performing authentication)
-    BeforeCompletion { generation: u32 },
+    /// Describes a point that occurred before the initial hello completed (e.g. when opening the
+    /// socket).
+    PreHello { generation: PoolGeneration },
+
+    /// Describes a point in time after the initial hello has completed, but before authentication.
```

Comment:
maybe instead of saying before authentication, we can say "before the entire handshake (e.g. including authentication) completes", since this phase includes errors due to authentication (and any other steps in the future).

Reply:
Done.